tests/test_assets.py

```diff
@@ -176,65 +209,89 @@ def build_lookup_generic_cases(asset_finder_type):
             adjustment=None,
         )
 
-        dupe_0_start = dupe_0.start_date
-        dupe_1_start = dupe_1.start_date
-        yield (
-            ##
-            # Scalars
-
-            # Asset object
-            (finder, assets[0], None, assets[0]),
-            (finder, assets[1], None, assets[1]),
-            (finder, assets[2], None, assets[2]),
-            # int
-            (finder, 0, None, assets[0]),
-            (finder, 1, None, assets[1]),
-            (finder, 2, None, assets[2]),
-            # Duplicated symbol with resolution date
-            (finder, 'DUPLICATED', dupe_0_start, dupe_0),
-            (finder, 'DUPLICATED', dupe_1_start, dupe_1),
-            # Unique symbol, with or without resolution date.
-            (finder, 'UNIQUE', unique_start, unique),
-            (finder, 'UNIQUE', None, unique),
-
-            # Futures
-            (finder, 'FOF14', None, fof14),
-            # Future symbols should be unique, but including as_of date
-            # make sure that code path is exercised.
-            (finder, 'FOF14', unique_start, fof14),
-
-            # Futures int
-            (finder, fof14_sid, None, fof14),
-            # Future symbols should be unique, but including as_of date
-            # make sure that code path is exercised.
-            (finder, fof14_sid, unique_start, fof14),
-
-            # ContinuousFuture
-            (finder, cf, None, cf),
-
-            ##
-            # Iterables
-
-            # Iterables of Asset objects.
-            (finder, assets, None, assets),
-            (finder, iter(assets), None, assets),
-            # Iterables of ints
-            (finder, (0, 1), None, assets[:-1]),
-            (finder, iter((0, 1)), None, assets[:-1]),
-            # Iterables of symbols.
-            (finder, ('DUPLICATED', 'UNIQUE'), dupe_0_start, [dupe_0, unique]),
-            (finder, ('DUPLICATED', 'UNIQUE'), dupe_1_start, [dupe_1, unique]),
-            # Mixed types
-            (finder,
-             ('DUPLICATED', 2, 'UNIQUE', 1, dupe_1),
-             dupe_0_start,
-             [dupe_0, assets[2], unique, assets[1], dupe_1]),
-
-            # Futures and Equities
-            (finder, ['FOF14', 0], None, [fof14, assets[0]]),
-
-            # ContinuousFuture and Equity
-            (finder, [cf, 0], None, [cf, assets[0]]),
+        all_assets = list(equities) + [fof14, cf]
+
+        dupe_old_start = dupe_old.start_date
+        dupe_new_start = dupe_new.start_date
+
+        for asset in list(equities) + [fof14, cf]:
+            # Looking up an asset object directly should yield itself.
+            yield case(asset, None, None, asset, )
+            # Looking up an asset by sid should yield the asset.
+            yield case(asset.sid, None, None, asset)
+
+        # Duplicated US equity symbol with resolution date.
+        for country in ('US', None):
+            yield case('DUPLICATED_IN_US', dupe_old_start, country, dupe_old)
```

Comment:
should we check a sample of days?

Reply:
I updated this to check the boundary condition of `old_start, new_start - minute, new_start, new_start + minute`. We already have very good coverage for `lookup_symbol`, which is ultimately what this function calls into, so I don't think it's important to check a sample of the interior here.

Reply:
makes sense